modules/host/net.go

```diff
@@ -22,6 +23,9 @@ func (h *Host) listen() {
 // TODO: maintain compatibility
 func (h *Host) handleConn(conn net.Conn) {
 	defer conn.Close()
+	// Set an initial duration that is generous, but finite. RPCs can extend
+	// this if so desired.
+	conn.SetDeadline(time.Now().Add(5 * time.Minute))
```

Comment:
does the ability to extend lead to a DoS vector?

Also, 5 minutes is 110 kbps for a 4MB file. But I think that's fast enough. Was thinking about people with dialup-grade upload speeds, but honestly if your upspeed is that slow you're not really going to be able to make much use of Sia anyway.